field/src/field.rs

```diff
@@ -195,7 +196,21 @@ pub trait AbstractExtensionField<Base>:
     fn as_base_slice(&self) -> &[Base];
 }
 
-pub trait ExtensionField<Base: Field>: Field + AbstractExtensionField<Base> {}
+pub trait ExtensionField<Base: Field>: Field + AbstractExtensionField<Base> {
+    fn is_in_basefield(&self) -> bool {
+        self.as_base_slice()[1..].iter().all(|x| x.is_zero())
+    }
+
+    fn scalar_mul(&self, scalar: Base) -> Self {
+        let value: &[Base] = self.as_base_slice();
+        let mut res = Vec::with_capacity(Self::D);
+
+        value.iter().for_each(|x| {
+            res.push(*x * scalar);
+        });
+        Self::from_base_slice(res.as_slice())
+    }
```

Comment:
This implementation using a `Vec` seems a bit inefficient when the extension field is of the form `EF([Base; D])`, like in the case of `QuadraticOef` and `CubicOef`. 
We could overwrite it in these cases, like
```rust
fn scalar_mul(&self, scalar: Base) -> Self {
        let mut res = self.0;
        res.iter_mut().for_each(|x| {
            *x *= scalar;
        });
        Self(res)
}
```